cmd/snap/cmd_validate.go

```diff
@@ -0,0 +1,193 @@
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"enforce": i18n.G("Enforce the given validation set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"forget": i18n.G("Forget the given validation set"),
+	}), []argDesc{{
+		// TRANSLATORS: This needs to begin with < and end with >
+		name: i18n.G("<validation-set>"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		desc: i18n.G("Validation set with an optional pinned sequence point, i.e. account/name[=seq]"),
+	}})
+}
+
+// this is reused for both account and set name of "account/name" argument
+var validName = regexp.MustCompile("^[a-z][0-9a-z]+$")
+
+func splitValidationSetArg(arg string) (account, name string, seq int, err error) {
+	parts := strings.Split(arg, "=")
+	if len(parts) > 2 {
+		return "", "", 0, fmt.Errorf("cannot parse validation set, expected account/name=seq")
+	}
+	if len(parts) == 2 {
+		seq, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return "", "", 0, err
+		}
+	}
+
+	parts = strings.Split(parts[0], "/")
+	if len(parts) != 2 {
+		return "", "", 0, fmt.Errorf("expected a single account/name")
+	}
+
+	account = parts[0]
+	name = parts[1]
+	if !validName.MatchString(account) {
+		return "", "", 0, fmt.Errorf("invalid account name %q", account)
+	}
+	if !validName.MatchString(name) {
+		return "", "", 0, fmt.Errorf("invalid name %q", name)
+	}
+
+	return account, name, seq, nil
+}
+
+func fmtValid(res *client.ValidationSetResult) string {
+	if res.Valid {
+		return fmt.Sprintf("valid")
+	}
+	return fmt.Sprintf("invalid")
+}
+
+func (cmd *cmdValidate) Execute(args []string) error {
+	// check that only one action is used at a time
+	var action string
+	for _, a := range []struct {
+		name string
+		set  bool
+	}{
+		{"monitor", cmd.Monitor},
+		{"enforce", cmd.Enforce},
+		{"forget", cmd.Forget},
+	} {
+		if a.set {
+			if action != "" {
+				return fmt.Errorf("cannot use --%s and --%s together", action, a.name)
+			}
+			action = a.name
+		}
+	}
+
+	if cmd.Positional.ValidationSet == "" && action != "" {
+		return fmt.Errorf("missing validation set argument")
+	}
+
+	var account, name string
+	var seq int
+	var err error
+	if cmd.Positional.ValidationSet != "" {
+		account, name, seq, err = splitValidationSetArg(cmd.Positional.ValidationSet)
+		if err != nil {
+			return fmt.Errorf("cannot parse validation set %q: %v", cmd.Positional.ValidationSet, err)
+		}
+	}
+
+	if action != "" {
+		// forget
+		if cmd.Forget {
+			return cmd.client.ForgetValidationSet(account, name, seq)
+		}
+		// apply
+		opts := &client.ValidateApplyOptions{
+			Mode:     action,
+			Sequence: seq,
+		}
+		return cmd.client.ApplyValidationSet(account, name, opts)
+	}
+
+	// no validation set argument, print list with extended info
+	if cmd.Positional.ValidationSet == "" {
+		vsets, err := cmd.client.ListValidationsSets()
+		if err != nil {
+			return err
+		}
+		if len(vsets) == 0 {
+			return nil
```

Comment:
Should we print something here? Like "No validations sets are defined yet" or something (similar what we do if we have no snaps in snap list)?

Reply:
Indeed, I missed it in the spec, we should print 'No validations are available.'. Will add, thanks!